elastic/apm-server
# Enable synthetic source for metrics data streams

## Motivation/summary

Enable synthetic source by default for APM's metrics data streams. See https://github.com/elastic/apm-server/issues/9010 for motiviation.

It is possible to disable synthetic source by modifying the `@custom` component template, e.g.

```
PUT /_component_template/metrics-apm.app@custom
{
  "template": {
    "mappings": {
      "_source": {}
    }
  }
}
```

## Checklist

~- [ ] Documentation has been updated~

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

1. Send some transactions, spans with `service.target.*`, and custom application metrics
2. Confirm that synthetic source is enabled for the APM metrics data streams (internal & app)
3. Confirm that transaction metrics and service dependencies are visualised in the APM UI
4. Confirm that application metrics can be visualised in Lens
5. Confirm that the (synthetic) source can be visualised for all events in Discover
6. Modify component templates for metrics data streams to disable synthetic source (see Summary section for instructions) and roll over; send more data and ensure source is not synthetic

## Related issues

Closes #9010
